<a href="https://colab.research.google.com/github/Douglas1997Ortiz/predicciones-de-ventas/blob/main/PROYECTO2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import LogisticRegression
import seaborn as sn 

In [ ]:
from google.colab import drive 
drive.mount('/content/drive')

Mounted at /content/drive


# PROPUESTA 1
# PRECIO DEL ORO

In [ ]:
df1= pd.read_csv('/content/drive/MyDrive/Coding Dojo/PROYECTO 2/PRECIO DEL ORO.csv')
df1.head()

Date        Open        High         Low       Close   Adj Close  \
0  2011-12-15  154.740005  154.949997  151.710007  152.330002  152.330002   
1  2011-12-16  154.309998  155.369995  153.899994  155.229996  155.229996   
2  2011-12-19  155.479996  155.860001  154.360001  154.869995  154.869995   
3  2011-12-20  156.820007  157.429993  156.580002  156.979996  156.979996   
4  2011-12-21  156.979996  157.529999  156.130005  157.160004  157.160004   

     Volume     SP_open     SP_high      SP_low  ...    GDX_Low  GDX_Close  \
0  21521900  123.029999  123.199997  121.989998  ...  51.570000  51.680000   
1  18124300  122.230003  122.949997  121.300003  ...  52.040001  52.680000   
2  12547200  122.059998  122.320000  120.029999  ...  51.029999  51.169998   
3   9136300  122.180000  124.139999  120.370003  ...  52.369999  52.990002   
4  11996100  123.930000  124.360001  122.750000  ...  52.419998  52.959999   

   GDX_Adj Close  GDX_Volume   USO_Open   USO_High    USO_Low  USO_Close  \
0      48.973877    20605600  36.900002  36.939999  36.049999  36.130001   
1      49.921513    16285400  36.180000  36.500000  35.730000  36.270000   
2      48.490578    15120200  36.389999  36.450001  35.930000  36.200001   
3      50.215282    11644900  37.299999  37.610001  37.220001  37.560001   
4      50.186852     8724300  37.669998  38.240002  37.520000  38.110001   

   USO_Adj Close  USO_Volume  
0      36.130001    12616700  
1      36.270000    12578800  
2      36.200001     7418200  
3      37.560001    10041600  
4      38.110001    10728000  

[5 rows x 81 columns]

In [ ]:
df1.columns

Index(['Date', 'Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume',
       'SP_open', 'SP_high', 'SP_low', 'SP_close', 'SP_Ajclose', 'SP_volume',
       'DJ_open', 'DJ_high', 'DJ_low', 'DJ_close', 'DJ_Ajclose', 'DJ_volume',
       'EG_open', 'EG_high', 'EG_low', 'EG_close', 'EG_Ajclose', 'EG_volume',
       'EU_Price', 'EU_open', 'EU_high', 'EU_low', 'EU_Trend', 'OF_Price',
       'OF_Open', 'OF_High', 'OF_Low', 'OF_Volume', 'OF_Trend', 'OS_Price',
       'OS_Open', 'OS_High', 'OS_Low', 'OS_Trend', 'SF_Price', 'SF_Open',
       'SF_High', 'SF_Low', 'SF_Volume', 'SF_Trend', 'USB_Price', 'USB_Open',
       'USB_High', 'USB_Low', 'USB_Trend', 'PLT_Price', 'PLT_Open', 'PLT_High',
       'PLT_Low', 'PLT_Trend', 'PLD_Price', 'PLD_Open', 'PLD_High', 'PLD_Low',
       'PLD_Trend', 'RHO_PRICE', 'USDI_Price', 'USDI_Open', 'USDI_High',
       'USDI_Low', 'USDI_Volume', 'USDI_Trend', 'GDX_Open', 'GDX_High',
       'GDX_Low', 'GDX_Close', 'GDX_Adj Close', 'GDX_Volume', 'USO_Open',
       'USO_Hig

In [ ]:
df1.shape

(1718, 81)

# Primera opción: conjunto de datos 1

# 1) Fuente de datos
 Kaggle

# 2) Breve descripción de datos
Los datos pertenecen a los precios de apertura, cierre, maximos, bajos, precio ajustado alcierre y volumen negociado por dia del Oro y los principales indices y acciones que influyen en el precio del oro

# 3) ¿Cuál es el objetivo?
Predecir el precio del Oro en relacion a los demas indices del mercado

# 4) ¿Este es un problema de clasificación o regresión?
Es un problema de regresion 

# 5) ¿Cuántas características hay?
81 columnas 

# 6) ¿Cuántas filas de datos hay?
1718 filas

# 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?
Retos, saber cual precio de las diferentes acciones es la que influyen en el precio del oro, imputar datos de precios de acciones volatiles.

# PROPUESTA 2
# SCORE BANKING

In [ ]:
df2_test= pd.read_csv('/content/drive/MyDrive/Coding Dojo/PROYECTO 2/TEST SCORE BANKING.csv')
dtype_dict = {'Monthly_Balance': str}
df2_train= pd.read_csv('/content/drive/MyDrive/Coding Dojo/PROYECTO 2/TRAIN SOCRE BANKING.csv' , dtype=dtype_dict)


In [ ]:
print(df2_train.columns[26])

Monthly_Balance


In [ ]:
df2_train.columns

Index(['ID', 'Customer_ID', 'Month', 'Name', 'Age', 'SSN', 'Occupation',
       'Annual_Income', 'Monthly_Inhand_Salary', 'Num_Bank_Accounts',
       'Num_Credit_Card', 'Interest_Rate', 'Num_of_Loan', 'Type_of_Loan',
       'Delay_from_due_date', 'Num_of_Delayed_Payment', 'Changed_Credit_Limit',
       'Num_Credit_Inquiries', 'Credit_Mix', 'Outstanding_Debt',
       'Credit_Utilization_Ratio', 'Credit_History_Age',
       'Payment_of_Min_Amount', 'Total_EMI_per_month',
       'Amount_invested_monthly', 'Payment_Behaviour', 'Monthly_Balance',
       'Credit_Score'],
      dtype='object')

In [ ]:
df2=pd.concat([df2_train, df2_train], axis=0)
df2.head()

ID Customer_ID     Month           Name   Age          SSN Occupation  \
0  0x1602   CUS_0xd40   January  Aaron Maashoh    23  821-00-0265  Scientist   
1  0x1603   CUS_0xd40  February  Aaron Maashoh    23  821-00-0265  Scientist   
2  0x1604   CUS_0xd40     March  Aaron Maashoh  -500  821-00-0265  Scientist   
3  0x1605   CUS_0xd40     April  Aaron Maashoh    23  821-00-0265  Scientist   
4  0x1606   CUS_0xd40       May  Aaron Maashoh    23  821-00-0265  Scientist   

  Annual_Income  Monthly_Inhand_Salary  Num_Bank_Accounts  ...  Credit_Mix  \
0      19114.12            1824.843333                  3  ...           _   
1      19114.12                    NaN                  3  ...        Good   
2      19114.12                    NaN                  3  ...        Good   
3      19114.12                    NaN                  3  ...        Good   
4      19114.12            1824.843333                  3  ...        Good   

   Outstanding_Debt Credit_Utilization_Ratio     Credit_History_Age  \
0            809.98                26.822620  22 Years and 1 Months   
1            809.98                31.944960                    NaN   
2            809.98                28.609352  22 Years and 3 Months   
3            809.98                31.377862  22 Years and 4 Months   
4            809.98                24.797347  22 Years and 5 Months   

   Payment_of_Min_Amount Total_EMI_per_month Amount_invested_monthly  \
0                     No           49.574949       80.41529543900253   
1                     No           49.574949      118.28022162236736   
2                     No           49.574949         81.699521264648   
3                     No           49.574949       199.4580743910713   
4                     No           49.574949      41.420153086217326   

                  Payment_Behaviour     Monthly_Balance Credit_Score  
0   High_spent_Small_value_payments  312.49408867943663         Good  
1    Low_spent_Large_value_payments  284.62916249607184         Good  
2   Low_spent_Medium_value_payments   331.2098628537912         Good  
3    Low_spent_Small_value_payments  223.45130972736786         Good  
4  High_spent_Medium_value_payments  341.48923103222177         Good  

[5 rows x 28 columns]

In [ ]:
df2.shape

(200000, 28)

# Segunda opción: conjunto de datos 2

# 1) Fuente de datos
Kaggle

# 2) Breve descripción de datos
Muestra los diferentes aspectos que tiene un cliente para saber su score bancario

# 3) ¿Cuál es el objetivo?
Determinar los aspectos mas influyentes en la calificacion crediticia de un cliente 

# 4) ¿Este es un problema de clasificación o regresión?
Es un problema de clasificacion 

# 5) ¿Cuántas características hay?
28 caracteristicas

# 6) ¿Cuántas filas de datos hay?
200.000 filas de datos

# 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?
La df, posee datos por transaccion, no por clientes, asi se puede tener que un cliente pueda tener 1 transaccion como 10 transacciones, caracteristicas como edad, credit mix, que son de caracter string y que no se pueden imputar facilmente.